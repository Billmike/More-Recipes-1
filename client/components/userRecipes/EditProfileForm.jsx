import React from 'react';
import PropTypes from 'prop-types';
import ReactTooltip from 'react-tooltip';

import TextField from '../common/TextField';

const EditProfileForm = (props) => {
  const { user: { userProfile }, handleProfileChange, handleSubmitProfile,
      handleCancel } = props,
    { username, firstName, lastName, email, aboutMe } = userProfile;
  return (
    <div>
      <div className="row" />
      <h5><i className="material-icons left">person</i>My Profile</h5>
      <div className="divider black" />
      <br />
      <div id="about_user" className="row">
        <TextField
          id="Username"
          name="username"
          type="text"
          placeholder="Username"
          defaultValue={username}
          className="validate"
          disabled
          onChange={handleProfileChange}
        />
        <TextField
          id="First Name"
          name="firstName"
          type="text"
          placeholder="First name"
          defaultValue={firstName}
          className="validate"
          required
          onChange={handleProfileChange}
        />
        <TextField
          id="Last Name"
          name="lastName"
          type="text"
          placeholder="Last name"
          defaultValue={lastName}
          className="validate"
          required
          onChange={handleProfileChange}
        />
        <TextField
          id="Email"
          name="email"
          type="email"
          placeholder="Email"
          defaultValue={email}
          disabled
          onChange={handleProfileChange}
        />
      </div>
      <div className="col s12">
        <h5>About Me</h5>
        <div className="divider black" />
        <textarea
          id="textarea"
          name="aboutMe"
          placeholder="About me"
          defaultValue={aboutMe}
          className="materialize-textarea"
          onChange={handleProfileChange}
        />
      </div>
      <div>
        <button
          id="edit-btn"
          className={`btn-floating waves-effect waves-light indigo accent-2
        right`}
          onClick={handleSubmitProfile}
        >
          <i className="material-icons" data-tip="Save details">save</i>
        </button>
        <button
          id="edit-btn"
          className={'btn-floating waves-effect waves-light red right'}
          onClick={handleCancel}
        >
          <i className="material-icons" data-tip="Cancel">cancel</i>
        </button>
        <ReactTooltip />
      </div>
      <div className="row" />
    </div>
  );
};

EditProfileForm.propTypes = {
  user: PropTypes.shape({
    userProfile: PropTypes.shape({
      username: PropTypes.string,
      firstName: PropTypes.string,
      lastName: PropTypes.string,
      email: PropTypes.string,
      aboutMe: PropTypes.string,
    })
  }).isRequired,
  handleProfileChange: PropTypes.func.isRequired,
  handleSubmitProfile: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired,
};

export default EditProfileForm;
