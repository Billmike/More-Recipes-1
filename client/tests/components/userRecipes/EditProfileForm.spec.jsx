import React from 'react';

import EditProfileForm from '../../../components/userRecipes/EditProfileForm';

describe('EditProfileForm component', () => {
  it('should render component', () => {
    const props = {
      user: {
        isLoading: false,
        userProfile: {
          username: 'John Mark',
          firstName: 'John',
          lastName: 'Mark',
          email: 'john@example.com',
          aboutMe: 'A lot about me',
        },
      },
      handleProfileChange: jest.fn(),
      handleSubmitProfile: jest.fn(),
      handleCancel: jest.fn(),
    };
    const wrapper = shallow(<EditProfileForm {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should call onChange handler function', () => {
    const props = {
      user: {
        isLoading: false,
        userProfile: {
          username: 'John Mark',
          firstName: 'John',
          lastName: 'Mark',
          email: 'john@example.com',
          aboutMe: 'A lot about me',
        },
      },
      handleProfileChange: jest.fn(),
      handleSubmitProfile: jest.fn(),
      handleCancel: jest.fn(),
    };
    const wrapper = shallow(<EditProfileForm {...props} />);

    expect(wrapper.find('#about_user').childAt(0)
      .props().disabled).toBeTruthy();
    wrapper.find('#about_user').childAt(1).simulate(
      'change', { target: { value: 'JOHN' } }
    );
    expect(props.handleProfileChange).toBeCalled();
    wrapper.find('#about_user').childAt(2).simulate(
      'change', { target: { value: 'MARK' } }
    );
    expect(props.handleProfileChange).toBeCalled();
    expect(wrapper.find('#about_user').childAt(3)
      .props().disabled).toBeTruthy();
    wrapper.find('textarea#aboutMe').simulate('change', {
      target: { value: 'You can hola me to know' }
    });
    expect(props.handleProfileChange).toBeCalled();
    wrapper.find('button#confirm-edit-btn').simulate('click');
    expect(props.handleSubmitProfile).toBeCalled();
    wrapper.find('button#cancel-edit-btn').simulate('click');
    expect(props.handleCancel).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
  it('should show spinner', () => {
    const props = {
      user: {
        isLoading: true,
        userProfile: {
          username: 'John Mark',
          firstName: 'John',
          lastName: 'Mark',
          email: 'john@example.com',
          aboutMe: 'A lot about me',
        },
      },
      handleProfileChange: jest.fn(),
      handleSubmitProfile: jest.fn(),
      handleCancel: jest.fn(),
    };

    const wrapper = shallow(<EditProfileForm {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
