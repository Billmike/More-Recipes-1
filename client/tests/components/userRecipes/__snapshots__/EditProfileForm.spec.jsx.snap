// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileForm component should call onChange handler function 1`] = `
<div>
  <div
    className="row"
  />
  <h5>
    <i
      className="material-icons left"
    >
      person
    </i>
    My Profile
  </h5>
  <div
    className="divider black"
  />
  <br />
  <div
    className="row"
    id="about_user"
  >
    <TextField
      className="validate"
      defaultValue="John Mark"
      disabled={true}
      id="Username"
      name="username"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "target": Object {
                  "value": "JOHN",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "MARK",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "You can hola me to know",
                },
              },
            ],
          ],
        }
      }
      placeholder="Username"
      type="text"
    />
    <TextField
      className="validate"
      defaultValue="John"
      id="First Name"
      maxLength="255"
      name="firstName"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "target": Object {
                  "value": "JOHN",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "MARK",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "You can hola me to know",
                },
              },
            ],
          ],
        }
      }
      placeholder="First name"
      required={true}
      type="text"
    />
    <TextField
      className="validate"
      defaultValue="Mark"
      id="Last Name"
      maxLength="255"
      name="lastName"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "target": Object {
                  "value": "JOHN",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "MARK",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "You can hola me to know",
                },
              },
            ],
          ],
        }
      }
      placeholder="Last name"
      required={true}
      type="text"
    />
    <TextField
      defaultValue="john@example.com"
      disabled={true}
      id="Email"
      name="email"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "target": Object {
                  "value": "JOHN",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "MARK",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "You can hola me to know",
                },
              },
            ],
          ],
        }
      }
      placeholder="Email"
      type="email"
    />
  </div>
  <div
    className="col s12"
  >
    <h5>
      About Me
    </h5>
    <div
      className="divider black"
    />
    <textarea
      className="materialize-textarea"
      defaultValue="A lot about me"
      id="aboutMe"
      name="aboutMe"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "target": Object {
                  "value": "JOHN",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "MARK",
                },
              },
            ],
            Array [
              Object {
                "target": Object {
                  "value": "You can hola me to know",
                },
              },
            ],
          ],
        }
      }
      placeholder="About me"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="center-align"
    />
  </div>
  <div>
    <button
      className="btn-floating waves-effect waves-light indigo accent-2
            right"
      disabled={false}
      id="confirm-edit-btn"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
    >
      <i
        className="material-icons"
        data-tip="Save Details"
      >
        save
      </i>
    </button>
    <button
      className="btn-floating waves-effect waves-light red right"
      disabled={false}
      id="cancel-edit-btn"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
    >
      <i
        className="material-icons"
        data-tip="Cancel"
      >
        cancel
      </i>
    </button>
    <ReactTooltip
      insecure={true}
      resizeHide={true}
      wrapper="div"
    />
  </div>
  <div
    className="row"
  />
</div>
`;

exports[`EditProfileForm component should render component 1`] = `
<div>
  <div
    className="row"
  />
  <h5>
    <i
      className="material-icons left"
    >
      person
    </i>
    My Profile
  </h5>
  <div
    className="divider black"
  />
  <br />
  <div
    className="row"
    id="about_user"
  >
    <TextField
      className="validate"
      defaultValue="John Mark"
      disabled={true}
      id="Username"
      name="username"
      onChange={[MockFunction]}
      placeholder="Username"
      type="text"
    />
    <TextField
      className="validate"
      defaultValue="John"
      id="First Name"
      maxLength="255"
      name="firstName"
      onChange={[MockFunction]}
      placeholder="First name"
      required={true}
      type="text"
    />
    <TextField
      className="validate"
      defaultValue="Mark"
      id="Last Name"
      maxLength="255"
      name="lastName"
      onChange={[MockFunction]}
      placeholder="Last name"
      required={true}
      type="text"
    />
    <TextField
      defaultValue="john@example.com"
      disabled={true}
      id="Email"
      name="email"
      onChange={[MockFunction]}
      placeholder="Email"
      type="email"
    />
  </div>
  <div
    className="col s12"
  >
    <h5>
      About Me
    </h5>
    <div
      className="divider black"
    />
    <textarea
      className="materialize-textarea"
      defaultValue="A lot about me"
      id="aboutMe"
      name="aboutMe"
      onChange={[MockFunction]}
      placeholder="About me"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="center-align"
    />
  </div>
  <div>
    <button
      className="btn-floating waves-effect waves-light indigo accent-2
            right"
      disabled={false}
      id="confirm-edit-btn"
      onClick={[MockFunction]}
    >
      <i
        className="material-icons"
        data-tip="Save Details"
      >
        save
      </i>
    </button>
    <button
      className="btn-floating waves-effect waves-light red right"
      disabled={false}
      id="cancel-edit-btn"
      onClick={[MockFunction]}
    >
      <i
        className="material-icons"
        data-tip="Cancel"
      >
        cancel
      </i>
    </button>
    <ReactTooltip
      insecure={true}
      resizeHide={true}
      wrapper="div"
    />
  </div>
  <div
    className="row"
  />
</div>
`;

exports[`EditProfileForm component should show spinner 1`] = `
<div>
  <div
    className="row"
  />
  <h5>
    <i
      className="material-icons left"
    >
      person
    </i>
    My Profile
  </h5>
  <div
    className="divider black"
  />
  <br />
  <div
    className="row"
    id="about_user"
  >
    <TextField
      className="validate"
      defaultValue="John Mark"
      disabled={true}
      id="Username"
      name="username"
      onChange={[MockFunction]}
      placeholder="Username"
      type="text"
    />
    <TextField
      className="validate"
      defaultValue="John"
      id="First Name"
      maxLength="255"
      name="firstName"
      onChange={[MockFunction]}
      placeholder="First name"
      required={true}
      type="text"
    />
    <TextField
      className="validate"
      defaultValue="Mark"
      id="Last Name"
      maxLength="255"
      name="lastName"
      onChange={[MockFunction]}
      placeholder="Last name"
      required={true}
      type="text"
    />
    <TextField
      defaultValue="john@example.com"
      disabled={true}
      id="Email"
      name="email"
      onChange={[MockFunction]}
      placeholder="Email"
      type="email"
    />
  </div>
  <div
    className="col s12"
  >
    <h5>
      About Me
    </h5>
    <div
      className="divider black"
    />
    <textarea
      className="materialize-textarea"
      defaultValue="A lot about me"
      id="aboutMe"
      name="aboutMe"
      onChange={[MockFunction]}
      placeholder="About me"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="center-align"
    >
      <Spinner
        size="small"
      />
    </div>
  </div>
  <div>
    <button
      className="btn-floating waves-effect waves-light indigo accent-2
            right"
      disabled={true}
      id="confirm-edit-btn"
      onClick={[MockFunction]}
    >
      <i
        className="material-icons"
        data-tip="Save Details"
      >
        save
      </i>
    </button>
    <button
      className="btn-floating waves-effect waves-light red right"
      disabled={true}
      id="cancel-edit-btn"
      onClick={[MockFunction]}
    >
      <i
        className="material-icons"
        data-tip="Cancel"
      >
        cancel
      </i>
    </button>
    <ReactTooltip
      insecure={true}
      resizeHide={true}
      wrapper="div"
    />
  </div>
  <div
    className="row"
  />
</div>
`;
